### DISCUSSION TITLE
Upgrade from Twill 1.0 to Twill 3.0

### DISCUSSION THREAD

**Discussion by Nandana-S** [area17/twill#2313] [Q&A]
How do you upgrade from Twill 1.0 to 3.0? The process for upgrading from 2.x to 3.0 was available on the website, but I could not find any documentation for upgrading from 1.0 to 2.0, or 1.0 to 3.0. Could anyone provide a link for the same?

**Comment by ifox** [comment]
Hi @Nandana-S sorry for the delay. 

1.0 to 2.0 was not a lot of breaking change. The changelog is the best resource for that upgrade, here: https://github.com/area17/twill/blob/2.x/CHANGELOG.md#200-2020-03-06
